versions/1/integration/elasticsearch/spec.yml

```diff
@@ -28,3 +28,48 @@ spec:
           pattern: '^.+\.json$'
           contentMediaType: "application/json"
           required: false
+    - description: Folder containing Elasticsearch Transforms
+      # https://www.elastic.co/guide/en/elasticsearch/reference/current/transforms.html
+      type: folder
+      name: transform
+      required: false
+      contents:
+        - description: |-
+            An Elasticsearch transform folder.
+            Upon installation of a package that contains a transform asset, the create behavior is as follows:
+            1. If a destination index template is included:
+              1. Create the destination index template.
+              2. Add the appropriate _meta fields to mark the matching indices as managed by Fleet.
+              3. If a fields directory is included:
+                1. Use the defined fields as the destination index template mappings.
+              4. if the destination index doesn't already exist:
+                1. Initialize the index.
+            3. Create the transform in the default namespace using kibana_system privileges.
+            5. Start the transform
+
+            Upon upgrading a package that contains a transform asset, the upgrade behavior is as follows:
+            1. If a destination index template is included:
+              1. Delete the destination index template.
+              2. Matching indices are not deleted.
+            2. Stop the transform.
+            3. Delete the transform.
+            4. From here, follow the creation behavior.
+          type: folder
+          pattern: '^[a-z0-9_]+$'
+          contents:
+            - description: Elasticsearch transform file
+              type: file
+              required: true
+              contentMediaType: "application/json"
+              pattern: '^transform\.json$'
+            - description: Elasticsearch index template for the transform's destination index
+              #  https://www.elastic.co/guide/en/elasticsearch/reference/current/index-templates.html
+              type: file
+              required: false
+              contentMediaType: "application/json"
+              pattern: '^dest_index_template\.json$'
```

Comment:
This wouldn't be needed now after adding the `fields` directory?

Reply:
I understand how the fields directory replaces the `mappings` part of the index template but I don't understand how the fields directory can replace some other parts of the destination index template, such as `aliases` for example. I'm probably missing something. Can you explain?
For example, for a user to create [this](https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-put-template.html#_index_template_with_index_aliases) index template (from the docs):
```
PUT _index_template/template_1
{
  "index_patterns" : ["te*"],
  "template": {
    "settings" : {
        "number_of_shards" : 1
    },
    "aliases" : {
        "alias1" : {},
        "alias2" : {
            "filter" : {
                "term" : {"user.id" : "kimchy" }
            },
            "routing" : "shard-1"
        },
        "{index}-alias" : {} 
    }
  }
}
```
How would the fields directory look like?

We can decide these parts of the index template are irrelevant for packages and not let them be included in the spec, but otherwise I don't understand how the fields directory is sufficient. 

Packages already can contain data stream assets which implicitly include index templates so I think the gap in my understanding here is how would a user be able to control these extra fields that aren't included in the fields directory (from what I understand).

Reply:
We would like to limit the features offered to what is needed. If we allow raw index templates, we would need to support all the combination of features available in index templates, we'd prefer to limit this to fields definitions if this is enough. Fields definition seem enough for the sample template included in this PR.

But as mentioned in an offline discussion, and looking to the real templates linked in https://github.com/elastic/package-spec/pull/307#issuecomment-1163003994, apart of the fields mappings definitions, we would also need to support some additional settings:
```
        "settings": {
            "index": {
                "codec": "best_compression",
                "refresh_interval": "5s",
                "number_of_shards": "1",
                "number_of_routing_shards": "30",
                "hidden": true
            }
        },
``` 
```
        "settings": {
            "index": {
                "codec": "best_compression",
                "refresh_interval": "5s",
                "number_of_shards": "1",
                "number_of_routing_shards": "30",
                "sort.field": [
                    "@timestamp",
                    "agent.id"
                ],
                "sort.order": [
                    "desc",
                    "asc"
                ]
            }
        },
```
This kind of settings are supported in data stream index templates, by using the [`elasticsearch.index_template.settings`](https://github.com/elastic/package-spec/blob/10b32dff0212c3c09970171f7702a77d4c3fd55c/versions/1/integration/data_stream/manifest.spec.yml#L198) setting in the data stream manifest.

Maybe an option is to add a manifest in the transform directory to include these settings, in a way that would be consistent with data streams.

@joshdover @mtojek what do you think is the best way to support these settings?

Reply:
@jsoriano I've updated the PR with your suggestion, and updated the examples based on the actual transforms and index templates used by the endpoint package